TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  AuthenticatedSecureStorage.swift
+//  TwilioSecurity
+//
+//  Copyright © 2021 Twilio.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import LocalAuthentication
+
+///:nodoc:
+public typealias SuccessBlock = (Data) -> ()
+///:nodoc:
+public typealias ErrorBlock = (Error) -> ()
+
+///:nodoc:
+public protocol AuthenticatedSecureStorageProvider {
+  func save(_ data: Data, withKey key: String, authenticator: Authenticator, success: @escaping EmptySuccessBlock, failure: @escaping ErrorBlock)
+  func get(_ key: String, authenticator: Authenticator, success: SuccessBlock, failure: ErrorBlock)
+  func removeValue(for key: String) throws
+  func clear() throws
+}
+
+///:nodoc:
+public class AuthenticatedSecureStorage {
+
+  private let keychain: KeychainProtocol
+  private let keychainQuery: KeychainQueryProtocol
+
+  public convenience init() {
+    self.init(keychain: Keychain(), keychainQuery: KeychainQuery())
+  }
+
+  init(keychain: KeychainProtocol = Keychain(),
+    keychainQuery: KeychainQueryProtocol = KeychainQuery()) {
```

Comment:
<table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/ffa803fef9e8654dd011341dfd173fb06285bdcd/TwilioVerifySDK/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorage.swift#L47"></span>Function parameters should be aligned vertically if they're in multiple lines in a declaration.<br />
<code>vertical_parameter_alignment</code> <code>AuthenticatedSecureStorage.swift:47</code></td>
    </tr>
  </tbody>
</table>